src/json/json5.h

```diff
@@ -26,13 +26,20 @@
 #include "util/location.h"
 #include "wcl/optional.h"
 #include "wcl/tracing.h"
+#include "wcl/unique_fd.h"
 
 class JsonSubscriber : public wcl::log::Subscriber {
  private:
-  std::ostream s;
+  wcl::unique_fd to_append;
+
+  explicit JsonSubscriber(wcl::unique_fd &&f) : to_append(std::move(f)) {}
 
  public:
-  JsonSubscriber(std::streambuf *rdbuf) : s(rdbuf) {}
+  JsonSubscriber(const JsonSubscriber &) = delete;
+  JsonSubscriber(JsonSubscriber &&) = default;
+  JsonSubscriber() = delete;
+  static wcl::result<JsonSubscriber, wcl::posix_error_t> create(const char *log_path);
```

Comment:
It's unclear to be why we need a static create function here.

Can you not just have the public api be  `explicit JsonSubscriber(wcl::unique_fd f)` then create the `unique_fd` where you create the `JsonSubscriber` then `std::move()` it in. Requiring a `path` here restricts the api (can't use stdin/stdout) and this feels like its doing more than it needs to

Reply:
Ah good point, you're right!

Reply:
It also makes error handling more specific to the context that its in

Reply:
Ok it turned out this was a bit more complex and we spoke off line. We don't want to alloy invalid constructions and it turns out there's only one valid way to construct the unique_fd so this does turn out to make sense

Reply:
Adding slightly more context here. I think the real problem was with the leaking error handling. I'm not going to block on it but ideally I'd like to see either the requirement encoded in the type or a less leaky error type returned